source/NVDAObjects/UIA/__init__.py

```diff
@@ -142,127 +224,203 @@ def _getControlFieldForObject(self, obj):
+				except UIAMixedAttributeError:
+					log.debug("Mixed formatting. Trying higher resolution unit")
+					for subfield in self._getTextWithFields_text(tempRange,formatConfig,UIAFormatUnits[1:]):
+						yield subfield
+					log.debug("Done yielding higher resolution unit")
+					continue
+				log.debug("Yielding formatting and text")
+				yield field
+				yield text
+		log.debug("Done _getTextWithFields_text")
+
+	def _getTextWithFieldsForUIARange(self,rootElement,textRange,formatConfig,includeRoot=False,alwaysWalkAncestors=True,recurseChildren=True,_rootElementRange=None):
+		"""
+		Yields start and end control fields, and text, for the given UI Automation text range.
+		@param rootElement: the highest ancestor that encloses the given text range. This function will not walk higher than this point.
+		@type rootElement: L{UIAHandler.IUIAutomation}
+		@param textRange: the UI Automation text range who's content should be fetched.
+		@type textRange: L{UIAHandler.IUIAutomation}
+		@param formatConfig: the types of formatting requested.
+		@ type formatConfig: a dictionary of NVDA document formatting configuration keys with values set to true for those types that should be fetched.
+		@param includeRoot: If true, then a control start and end will be yielded for the root element.
+		@ type includeRoot: bool
+		@param alwaysWalkAncestors: If true then control fields will be yielded for any element enclosing the given text range, that is a descendant of the root element. If false then the root element may be  assumed to be the only ancestor.
+		@type alwaysWalkAncestors: bool
+		@param recurseChildren: If true, this function will be recursively called for each child of the given text range, clipped to the bounds of this text range. Formatted text between the children will also be yielded. If false, only formatted text will be yielded.
+		@type recurseChildren: bool
+		@param _rootElementRange: Optimization argument: the actual text range for the root element, as it is usually already known when making recursive calls.
+		@type rootElementRange: L{UIAHandler.IUIAutomationTextRange} 
+		@rtype: A generator that yields L{textInfo.FieldCommand} objects and text strings.
+		"""
+		
+		if log.isEnabledFor(log.DEBUG):
+			log.debug("_getTextWithFieldsForUIARange")
+			log.debug("rootElement: %s"%rootElement.currentLocalizedControlType if rootElement else None)
+			log.debug("full text: %s"%textRange.getText(-1))
+		if recurseChildren:
+			childElements=textRange.getChildren()
+			# Specific check for embedded elements (checkboxes etc)
+			# Calling getChildren on their childRange always gives back the same child.
+			if childElements.length==1:
+				childElement=childElements.getElement(0)
+				if childElement and UIAHandler.handler.clientObject.compareElements(childElement,rootElement):
+					log.debug("Detected embedded child. Handling with _getTextWithFields_embedded")

```

Comment:
`_getTextWithFields_embedded` no longer exists. Instead, this gets handled below and there's another debug log there. Perhaps just cut the "Handling with..." bit of this message?